### ISSUE TITLE
podman: Too many open files

### ISSUE THREAD

**Issue by Loader009** [matrix-org/synapse#10462]
### Description

podman: Too many open files
Getting that error message in the homeserver.log with two different messages.
Error messages at the end.

py-spy doesn't work (permission denied), as mentioned here -> https://github.com/matrix-org/synapse/issues/2800#issuecomment-884657637
Even as root and with kernel.yama.ptrace_scope set to 0 or 1.

### Steps to reproduce

- start the container
- enter some bigger room
- wait

### Version information

- **Homeserver**: hidingin.space

- **Version**: v1.38.1

- **Install method**:
podman-compose -> docker-compose.yml
podman-rootless

- **Platform**:

podman version 3.1.2
Fedora 33 with selinux (completely updated and rebooted before the bugreport)

CPU: Intel Atom
RAM: 4G

That's the important part about hardware, it is enough to run synapse.
I will make worker threads later, to outsource intensive operations.

/etc/security/limits.conf:
```shell
# /etc/security/limits.conf
# […]

container   soft    nofile    65536
container   hard    nofile    65536

# End of file
```
docker-compose.yml:
```yaml
version: "3.9"

services:
  synapse:
#    image: docker.io/matrixdotorg/synapse:latest
    image: docker.io/matrixdotorg/synapse:v1.38.1
    stop_grace_period: 90s
    environment:
    - TZ=Europe/Berlin
    - SYNAPSE_SERVER_NAME=hidingin.space
    - SYNAPSE_REPORT_STATS=yes
    volumes:
    - ./data:/data
    security_opt:
    - label=disable
    network_mode: host
    ulimits:
      nofile:
        soft: 40000
#        hard: 40000
```
rootless user:
```bash
[container@hidingin.space synapse]$ ulimit -n
65536
```
Within the container:
```bash
$ podman exec -ti synapse_synapse_1 bash
root@hidingin.space:/# ulimit -n
40000
```
```syslog
2021-07-22 19:41:23,373 - twisted - 258 - CRITICAL - sentinel - Unhandled error in Deferred:
2021-07-22 19:41:23,374 - twisted - 258 - CRITICAL - sentinel - 
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/site-packages/twisted/python/threadpool.py", line 238, in inContext
    result = inContext.theWork()  # type: ignore[attr-defined]
  File "/usr/local/lib/python3.8/site-packages/twisted/python/threadpool.py", line 254, in <lambda>
    inContext.theWork = lambda: context.call(  # type: ignore[attr-defined]
  File "/usr/local/lib/python3.8/site-packages/twisted/python/context.py", line 118, in callWithContext
    return self.currentContext().callWithContext(ctx, func, *args, **kw)
  File "/usr/local/lib/python3.8/site-packages/twisted/python/context.py", line 83, in callWithContext
    return func(*args, **kw)
  File "/usr/local/lib/python3.8/site-packages/twisted/internet/_resolver.py", line 141, in get
    return self._getaddrinfo(
  File "/usr/local/lib/python3.8/socket.py", line 918, in getaddrinfo
    for res in _socket.getaddrinfo(host, port, family, type, proto, flags):
OSError: [Errno 24] Too many open files
```
```syslog
2021-07-22 19:41:23,837 - synapse.http.server - 93 - ERROR - GET-1287 - Failed handle request via 'ThumbnailResource': <XForwardedForRequest at 0x7f365c830340 method='GET' uri='/_matrix/media/r0/thumbnail/fairydust.space/QEDzjGtUNzXRxpBaLalwAWKD?width=14&height=14&method=crop' clientproto='HTTP/1.0' site='8008'>
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/site-packages/synapse/http/server.py", line 258, in _async_render_wrapper
  File "/usr/local/lib/python3.8/site-packages/synapse/http/server.py", line 286, in _async_render
  File "/usr/local/lib/python3.8/site-packages/synapse/rest/media/v1/thumbnail_resource.py", line 83, in _async_render_GET
  File "/usr/local/lib/python3.8/site-packages/synapse/rest/media/v1/thumbnail_resource.py", line 265, in _respond_remote_thumbnail
  File "/usr/local/lib/python3.8/site-packages/synapse/rest/media/v1/thumbnail_resource.py", line 321, in _select_and_respond_with_thumbnail
  File "/usr/local/lib/python3.8/site-packages/synapse/rest/media/v1/media_storage.py", line 189, in fetch_media
OSError: [Errno 24] Too many open files: '/data/media_store/remote_thumbnail/fairydust.space/CE/Zo/IqPJoQAVPXpZAifkHpzp/32-32-image-jpeg-crop'
```

**Comment by Loader009** [comment]
Just so that it's clear what limit it reaches:
![screen-2021-07-23-18-37-36](https://user-images.githubusercontent.com/9888311/126814012-71ecb7b0-24fa-4078-92d3-a0957fad2514.jpg)

![screen-2021-07-23-18-37-32](https://user-images.githubusercontent.com/9888311/126814015-d53fcc35-fee3-4b37-8c3f-6d5ce6ac40cd.jpg)

**Comment by erikjohnston** [comment]
Can you check your logs for a line of `Set file limit to:`, which happens just after start up and is what the Synapse process is setting the FD limit to, please?

**Comment by Loader009** [comment]
You were right, thank you!
That line was set to 200.
I didn't remember the setting inside homserver.yaml, but checked afterwards - it was indeed set to 200.
Sorry, this was my fault >.<
After I commented my setting out, it was at 40000, just what I would've expected.

My suggestion would be, to set the (commented) default higher (1.5k or 2k) but I can't find the default homeserver.yaml in the repo right now - I guess I'm just too tired today.
But maybe I had set this setting falsely to 200.
Sorry again, I close this.

PS: My homeserver barely touched 1250 FDs.
